File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/views/utils/image_controller/ImageControllerFactory.java
package coolpharaoh.tee.speicher.tea.timer.views.utils.image_controller;

import android.content.Context;

public class ImageControllerFactory {

    private ImageControllerFactory() {
    }

    private static ImageController mockedImageController;

    public static ImageController getImageController(final Context context) {
        if (mockedImageController != null) {
            return mockedImageController;
        } else {
            return new ContentResolverImageController(context);
        }
    }

    public static void setMockedImageController(final ImageController mockedImageController) {
        ImageControllerFactory.mockedImageController = mockedImageController;
    }
}

File: app/src/main/java/coolpharaoh/tee/speicher/tea/timer/views/utils/image_controller/ImageControllerFactory.kt
package coolpharaoh.tee.speicher.tea.timer.views.utils.image_controller

import android.content.Context

object ImageControllerFactory {

    private var mockedImageController: ImageController? = null

    @JvmStatic
    fun getImageController(context: Context): ImageController {
        return mockedImageController ?: ContentResolverImageController(context)
    }

    @JvmStatic
    fun setMockedImageController(mockedImageController: ImageController) {
        ImageControllerFactory.mockedImageController = mockedImageController
    }
}